packages/devtools/cli/src/util/credential.ts

```diff
@@ -0,0 +1,38 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { Trigger } from '@dxos/async';
+import { Client } from '@dxos/client';
+import { Credential } from '@dxos/protocols/proto/dxos/halo/credentials';
+
+const timeout = 500;
+
+export const queryCredentials = async (
+  client: Client,
+  type?: string,
+  predicate?: (value: Credential) => boolean
+): Promise<Credential[]> => {
+  const credentialsQuery = client.halo.queryCredentials({ type });
+  const trigger = new Trigger<Credential[]>();
+  let result: Credential[] = [];
+  credentialsQuery.subscribe({
+    onUpdate: (credentials: Credential[]) => {
+      if (predicate) {
+        result = credentials.filter(predicate);
+      } else {
+        result = credentials;
+      }
+    },
+    onError: (err) => {
+      throw err;
+    }
+  });
+  setTimeout(() => {
```

Comment:
Why is this required?

Reply:
@wittjosiah should this be halo.credentials.get?

Reply:
this is required because we are reading creds from feed, and they are coming progressively (not all at once).
e.g. if you have 5 creds in your halo, you will get 5 `onUpdate` events:
1 - with 1 cred
2 - with 2 creds
...
5 - with all 5 creds.
Since we don't really know how many credentials will be there, we could only wait for some time till we think all of them are returned

Reply:
We are working on this with @dmaretskyi, so for now timeout is ok